### PR TITLE
ResNet TF2 porting

### PR DESCRIPTION
**Pull request by ydudin3**
# Code Pull Requests

Please provide the following:

- a clear explanation of what your code dose
- if applicable, a reference to an issue
- a reproducible test for your PR (code, model definition and data sample)

# Documentation Pull Requests

Note that the documentation HTML files are in `docs/` while the Markdown sources are in `mkdocs/docs`.

If you are proposing a modification to the documentation you should change only the Markdown files.

`api.md` is automatically generated from the docstrings in the code, so if you want to change something in that file, first modify `ludwig/api.py` docstring, then run `mkdocs/code_docs_autogen.py`, which will create `mkdocs/docs/api.md` .

### CODE REVIEW

File: ludwig/models/modules/image_encoders.py [hunk 1]
```diff
@@ -187,36 +189,21 @@ def __init__(
...
         # ================ Conv Layers ================
-        hidden = self.resnet(
-            input_image,
-            regularizer,
-            dropout,
-            is_training=is_training
-        )
-        hidden = self.flatten(hidden)
+        hidden = self.resnet(inputs, training)
+        hidden = tf.reshape(hidden, [hidden.shape[0], -1])
```
Comment (w4nderlust) [inline ludwig/models/modules/image_encoders.py:204]: there's a tf.keras.layers.Flatten if you prefer to use that, but it does the same thing really

File: ludwig/models/modules/convolutional_modules.py [hunk 2]
```diff
@@ -722,6 +723,152 @@ def call(self, inputs, training=None, mask=None):
...
+        weights_regularizer=None,
+        batch_norm_momentum=0.9,
+        batch_norm_epsilon=0.001
+    ):
+        super(ResNet2, self).__init__()
+        self.resnet_size = resnet_size
+
+        self.is_bottleneck = is_bottleneck
+        if is_bottleneck:
+            self.block_fn = ResNetBottleneckBlock
```
Comment (w4nderlust) [inline ludwig/models/modules/convolutional_modules.py:801]: it seems that \block_fn\ is used only in this function so maybe we don't neeed to assign it to self. Also, as it's not a function anymore but a class, maybe let's cal it \block_class\?

**Comment by w4nderlust** [comment]
Thanks, this looks good! Now just make sure all dimensions and all inputs and outputs of every step is the right shape by comparing 1-1 lines of code with the previous implementantation. If everything is correct, we can move forward and merge.

File: ludwig/models/modules/convolutional_modules.py [hunk 3]
```diff
@@ -519,6 +524,349 @@ def call(self, inputs, training=None, mask=None):
...
+            filter_size=3,
+            strides=1,
+            weights_regularizer=None
+    ):
+        super(Conv2DLayerFixedPadding, self).__init__()
+
+        self.layers = []
+
+        if strides > 1:
+            self.layers.append(ZeroPadding2D(padding=((filter_size - 1) // 2)))
```
Comment (ydudin3) [inline ludwig/models/modules/convolutional_modules.py:541]: @w4nderlust have a look at this one, does this padding size make sense compared to the previous implementation?
